### PR TITLE
chore: enable circleci cron and cover src/**/*.ts files only

### LINKED ISSUE #645: Set up cron job on CircleCI

### PR DESCRIPTION
**Pull request by kjin**
__R__

* Enable nightly runs on CircleCI (fixes #645)
* Swap `istanbul` for `nyc`
* Report coverage for `src/**/*.ts` only (previously, it was `build/{src, test}/**/*.js`)
* Remove `GCLOUD_PROJECT` environmental variable (was interfering with system tests)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,7 +46,7 @@
   "author": "Google Inc.",
   "license": "Apache-2.0",
   "engines": {
-    "node": ">=0.12"
+    "node": ">=4"
```
Comment (ofrobots) [inline package.json]: Strictly speaking this is semver-major. Please include rationale for why this can be done without a major bump.

File: package.json [hunk 2]
```diff
@@ -72,12 +72,13 @@
     "google-auto-auth": "^0.9.0",
     "got": "^7.1.0",
     "gts": "^0.5.1",
-    "istanbul": "^0.4.2",
+    "istanbul": "^0.4.5",
```
Comment (ofrobots) [inline package.json]: Do we still need istanbul?

**Comment by ofrobots** [comment]
The coverage report seems empty. What's going on?

**Comment by kjin** [comment]
@ofrobots The paths were off. I fixed this by adding `codecov.yml`. PTAL

File: .circleci/config.yml [hunk 3]
```diff
@@ -78,6 +77,19 @@ mysql_service: &mysql_service
 version: 2.0
 workflows:
   version: 2
+  nightly:
+    triggers:
+      - schedule:
+          cron: "0 6 * * *" # Daily at 6:00AM
```
Comment (ofrobots) [inline .circleci/config.yml]: qq: do we know what timezone?

File: codecov.yml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+fixes:
+  - "root/project::"
```
Comment (ofrobots) [inline codecov.yml]: nit: add new line.

For my education; why was this needed?

**Comment by ofrobots** [comment]
Link to coveralls: https://coveralls.io/github/GoogleCloudPlatform/cloud-trace-nodejs

File: .circleci/config.yml (hunk 3 above)
Comment (kjin) [reply .circleci/config.yml]: It's UTC. I didn't think about this until you asked 🙃 

which makes it 10:00PM PT. I think that's reasonable since it's outside of normal work hours.

File: codecov.yml (hunk 4 above)
Comment (kjin) [reply codecov.yml]: When reported from CircleCI the paths are prefixed with `root/project` making it so that they couldn't be found relative to the root of the repository. This replaces that per https://docs.codecov.io/v4.3.6/docs/fixing-paths

**Comment by kjin** [comment]
@ofrobots Yep, the 96% seems to correspond to coverage in the latest open PR (#654), not this one.

**Comment by kjin** [comment]
Accidentally restarted AppVeyor CI even though it already passed. https://ci.appveyor.com/project/kjin/cloud-trace-nodejs/build/1.0.2102
